threedi_api_client/files.py

```diff
@@ -151,3 +153,112 @@ def download_fileobj(
+            requests (HTTP 429), service unavailable (HTTP 503)
+    """
+    # cast string to Path if necessary
+    if isinstance(file_path, str):
+        file_path = Path(file_path)
+
+    # open the file
+    with file_path.open("rb") as fileobj:
+        size = upload_fileobj(url, fileobj, timeout=timeout, pool=pool)
+
+    return size
+
+
+def upload_fileobj(
+    url: str,
+    fileobj: BinaryIO,
+    timeout: float = 5.0,
+    pool: Optional[urllib3.PoolManager] = None,
+) -> int:
+    """Upload a file object to a url.
+
+    The upload is accompanied by an MD5 hash so that the file server can check
+    the integrity of the file. This function is not inteded for files larger than
+    1 GB.
+
+    Args:
+        url: The url to upload to.
+        fileobj: The (binary) file object to read from.
+        size: The size of the file, in bytes.
+        timeout: The total timeout in seconds.
+        pool: If not supplied, a default connection pool will be
+            created with a retry policy of 3 retries after 1, 2, 4 seconds.
+
+    Returns:
+        The total amount of uploaded bytes.
+
+    Raises:
+        IOError: Raised if the provided file is incompatible or empty.
+        openapi_client.exceptions.ApiException: raised on unexpected server
+            responses (HTTP status codes other than 206, 413, 429, 503)
+        urllib3.exceptions.HTTPError: various low-level HTTP errors that persist
+            after retrying: connection errors, timeouts, decode errors,
+            invalid HTTP headers, payload too large (HTTP 413), too many
+            requests (HTTP 429), service unavailable (HTTP 503)
+    """
+    # There are two ways to upload in S3 (Minio):
+    # - PutObject: put the whole object in one time
+    # - multipart upload: requires presigned urls for every part
+    # We can only do the first option as we have no other presigned urls.
+
+    # We could do a streaming upload instead by supplying just the file stream as a
+    # body. Urllib3 will in that case forward the stream to http.HTTPConnection.request,
+    # which will stream using a rather small blocksize of 8192. This is hardcoded,
+    # see https://github.com/urllib3/urllib3/issues/2067
+
+    # Also if we would do that, we would have to pass through the file 2 times, once
+    # for computing the MD5 hash, once for the upload. So we just read the thing in memory:
+    body = fileobj.read()
+    if not isinstance(body, bytes):
+        raise IOError(
+            "The file object is not in binary mode. Please open with mode='rb'."
+        )
+    length = len(body)
+    if length == 0:
+        raise IOError("The file object is empty.")
+
+    # Make a hash so that the file server can check integerity.
+    md5 = base64.b64encode(hashlib.md5(body).digest())
+
+    if pool is None:
+        pool = get_pool()
+
+    headers = {"Content-Length": str(length), "Content-MD5": md5.decode()}
```

Comment:
Where did you find this? I cannot really find it in the Minio docs. Did you try if it works? ;)

Reply:
Yes I tried and it works (both content-length and content-md5). I basically looked at the Amazon S3 docs, I can't seem to find my way in the Minio docs